packages/devtools_app/lib/src/logging/flutter/logging_screen.dart

```diff
@@ -37,38 +46,77 @@ class LoggingScreenBody extends StatefulWidget {
 
 class _LoggingScreenState extends State<LoggingScreenBody>
     with AutoDisposeMixin {
-  LoggingController get controller => Controllers.of(context).logging;
   LogData selected;
+  TextEditingController filterController;
+
+  LoggingController get controller => Controllers.of(context).logging;
+
+  @override
+  void initState() {
+    super.initState();
+
+    filterController = TextEditingController();
+  }
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
+
     cancel();
-    addAutoDisposeListener(controller.onLogsUpdated);
+
+    filterController.text = controller.filterText;
+    filterController.addListener(() {
+      controller.filterText = filterController.text;
+    });
+
+    addAutoDisposeListener(controller.onLogsUpdated, () {
+      setState(() {
+        if (selected != null) {
+          final List<LogData> items = controller.filteredData;
+          if (!items.contains(selected)) {
+            selected = null;
+          }
+        }
+      });
+    });
   }
 
   @override
   Widget build(BuildContext context) {
     return Column(children: [
       Row(
-        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        mainAxisAlignment: MainAxisAlignment.start,
         children: [
           RaisedButton(
             child: const Text('Clear logs'),
             onPressed: _clearLogs,
           ),
+          const Spacer(),
+          Container(
+            width: 200.0,
+            height: 36.0,
```

Comment:
do you need to specify height: 36? Can you achieve the same effect by specifying cross-axis-alignment.

Reply:
I'll try this a few different ways.

Reply:
The default height of the text field is too large - it looks out of place with the other elements in the toolbar.